dbms/src/Storages/tests/gtest_SplitTokenExtractor.cpp

```diff
@@ -0,0 +1,174 @@
+#include <Storages/MergeTree/MergeTreeIndexFullText.h>
+
+#include <Common/PODArray_fwd.h>
+#include <Common/PODArray.h>
+
+#include <gtest/gtest.h>
+
+#include <vector>
+#include <string_view>
+#include <string>
+#include <memory>
+
+namespace
+{
+using namespace DB;
+}
+
+struct SplitTokenExtractorTestCase
+{
+    const char * description;
+    const std::string source;
+    const std::vector<std::string> tokens;
+};
+
+std::ostream & operator<<(std::ostream & ostr, const SplitTokenExtractorTestCase & test_case)
+{
+    return ostr << test_case.description;
+}
+
+class SplitTokenExtractorTest : public ::testing::TestWithParam<SplitTokenExtractorTestCase>
+{
+public:
+    void SetUp() override
+    {
+        const auto & param = GetParam();
+        const auto & source = param.source;
+        data = std::make_unique<PaddedPODArray<const char>>(source.data(), source.data() + source.size());
+    }
+
+    std::unique_ptr<PaddedPODArray<const char>> data;
+};
+
+TEST_P(SplitTokenExtractorTest, next)
+{
+    const auto & param = GetParam();
+
+    SplitTokenExtractor token_extractor;
+
+    size_t i = 0;
+
+    size_t pos = 0;
+    size_t token_start = 0;
+    size_t token_len = 0;
+
+    for (const auto & expected_token : param.tokens)
+    {
+        SCOPED_TRACE(++i);
```

Comment:
Why not SCOPED_TRACE(description + " " + token), I don't see how numbers make the life easier in detecting the test case

Reply:
since description and token are shown anyway in failed tests report, in case of similar tokens it pays to know what was the token index.

Reply:
Ok